js/prestashop-ui-kit.js

```diff
@@ -118,6 +118,27 @@ $.fn.pstooltip = $.fn.tooltip;
       });
     });
   }
+  
+  var initInputFileChangeEvent = function () {
+    jQuery('.custom-file input').on('change', function () {
+      var $input = $(this);
+      var files = $input[0].files;
+      var filesCount = files.length;
+      var $label = $input.next('label');
+
+      if (1 === filesCount) {
+        $label.text(files[0].name);
+      } else if (1 < filesCount) {
+        var multipleFilesTranslationText = $input.attr('data-multiple-files-text');
+
+        if (typeof multipleFilesTranslationText !== 'undefined') {
+          $label.text(filesCount + ' ' + multipleFilesTranslationText);
+        } else {
+          $label.text(filesCount + ' files');
```

Comment:
Consider extracting the logic that decides what text you'll put into your label, so that you end up having:

```js
$label.text(getLabelText(files))
```